providers/tpm2/bin/tpm2_capabilities.py

```diff
@@ -87,8 +87,9 @@
 try:
     algs_caps = subprocess.check_output(['tpm2_getcap', 'algorithms'])
     pcrs_caps = subprocess.check_output(['tpm2_getcap', 'pcrs'])
-except subprocess.CalledProcessError:
-    raise SystemExit
+except (subprocess.CalledProcessError, FileNotFoundError):
+    raise SystemExit("Please make sure you have installed tpm-tools or tpm chip")
```

Comment:
Definitely you didn't mean "or" here.

Reply:
If you have checked  [this submission](https://certification.canonical.com/hardware/202310-32222/submission/339052/test/129021/result/37058636/), you could see [clevis-encrypt-tpm2/detect-ecc-capabilities](https://certification.canonical.com/hardware/202310-32222/submission/339052/test/129020/result/37058635/) is passed while no tpm chip be detected and shouldn't be passed. This bug has been found by ODM too.

Reply:
and we use manifest to set TPM is installed or not in the DUT to prevent firmware or hardware issue to skip this test. The submission I provided points to a [real issue](https://bugs.launchpad.net/somerville/+bug/2040205) from the kernel.

Reply:
So both are needed, right?